### PR TITLE
feat(upload-sinan): refactor SINAN file upload

### PR DESCRIPTION
**Pull request by luabida**
depends on https://github.com/AlertaDengue/AlertaDengue/pull/692

### CODE REVIEW

**Review by fccoelho — APPROVED** [review]

File: AlertaDengue/upload/models.py [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+            "moved errored rows to `error_residue` file. "
+            "This error doesn't change the status to ERROR"
+        ))
+    error_residue = models.FileField(
+        null=True,
+        default=None,
+        help_text=_(
+            "Absolute CSV file path containing failed rows from data parsing, "
+            "before being uploaded to database. The filename format format is "
+            "RESIDUE_{filename} and it requires further human verification"
```
Comment (fccoelho) [inline AlertaDengue/upload/models.py:132]: Replace "residue" by "misparsed". Look for other places where it occurs, and replace as well.

File: AlertaDengue/upload/models.py [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+        choices=Status.choices,
+        default=Status.WAITING_CHUNK,
+        help_text=_("Upload status of the file")
+    )
+    parse_error = models.BooleanField(
+        null=False,
+        default=False,
+        help_text=_(
+            "An parse error ocurred when reading data, "
+            "moved errored rows to `error_residue` file. "
```
Comment (fccoelho) [inline AlertaDengue/upload/models.py:123]: change this line to:  "moved rows that failed parsing to `error_misparsed` file. "

File: AlertaDengue/upload/sinan/validations.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            raise ValidationError(_(f'File {file_path} not found'))
+
+
+def validate_residue_file_name(file_path: str) -> None:
+    if file_path:
+        fpath = Path(file_path)
+
+        if not fpath.name.startswith("RESIDUE_"):
+            raise ValidationError(_(
+                f"Residue file name {fpath.name} doesn't start with RESIDUE_"
```
Comment (fccoelho) [inline AlertaDengue/upload/sinan/validations.py:35]: should be: f"Misparsed file name... MISPARSED_"

**Review by fccoelho — CHANGES_REQUESTED** [review]

**Comment by luabida** [comment]
merging it to test in staging, the PR does not affect any other app

**Pull request merged**